test/e2e/broker_default_test.go

```diff
@@ -219,6 +222,44 @@ func TestDefaultBrokerWithManyTriggers(t *testing.T) {
 	}
 }
 
+// This test is for avoiding regressions on the trigger dependency annotation functionality.
+// It will first create a trigger with the dependency annotation, and then create a pingSource.
+// Broker controller should make trigger become ready after pingSource is ready.
+func TestTriggerDependencyAnnotation(t *testing.T) {
+	const (
+		triggerName          = "trigger-annotation"
+		subscriberName       = "subscriber-annotation"
+		dependencyAnnotation = `{"kind":"PingSource","name":"test-ping-source-annotation","apiVersion":"sources.knative.dev/v1alpha1"}`
+		pingSourceName       = "test-ping-source-annotation"
+		schedule             = "*/2 * * * *"
+	)
+	client := setup(t, true)
+	defer tearDown(client)
+
+	// Label namespace so that it creates the default broker.
+	if err := client.LabelNamespace(map[string]string{"knative-eventing-injection": "enabled"}); err != nil {
+		t.Fatalf("Error annotating namespace: %v", err)
+	}
+	// Wait for default broker ready.
+	client.WaitForResourceReadyOrFail(defaultBrokerName, lib.BrokerTypeMeta)
+
+	// Create triggers.
+	client.CreateTriggerOrFail(triggerName, resources.WithSubscriberServiceRefForTrigger(subscriberName), resources.WithDependencyAnnotaionTrigger(dependencyAnnotation))
+
+	pingSource := eventingtesting.NewPingSource(
+		pingSourceName,
+		client.Namespace,
+		eventingtesting.WithPingSourceSpec(sourcesv1alpha1.PingSourceSpec{
+			Sink:     &duckv1.Destination{Ref: resources.KnativeRefForService(defaultBrokerName, client.Namespace)},
+			Schedule: schedule,
+		}),
+	)
+	client.CreatePingSourceOrFail(pingSource)
+
+	// Trigger should become ready after pingSource was created
+	client.WaitForResourceReadyOrFail(triggerName, lib.TriggerTypeMeta)
```

Comment:
Did we check whether the trigger works or not? It looks like we only check the trigger can be ready or not? Did we check the whether the functionality of trigger works or not?

Reply:
I think the functionality of trigger works with broker is already checked in the other e2e tests (like TestDefaultBrokerWithManyTriggers). According to https://github.com/knative/eventing/issues/2582, my understanding for this E2E test is to check if the trigger can detect source changes with DependencyAnnotation. Am I missing something?

Reply:
It might be better to a e2e to test the functionality of the trigger with dependency annotation.

Reply:
Sure, no need to add another issue, I'll add it in this PR.